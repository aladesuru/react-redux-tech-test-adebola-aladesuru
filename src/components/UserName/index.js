import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const UserName = ({ firstName, lastName, userId }) => {
  return <Link to={`/userId/${userId}`}>{`${firstName} ${lastName}`}</Link>;
};

UserName.propTypes = {
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  userId: PropTypes.string.isRequired,
};

export default UserName;
