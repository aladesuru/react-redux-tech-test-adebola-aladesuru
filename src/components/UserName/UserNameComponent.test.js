import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import UserName from './index';

const renderUserNameComponent = (renderType) => {
  const requiredProps = {
    firstName: 'Henry',
    lastName: 'Harrison',
    userId: 'e837c9f5-247f-445f-bcc3-7d434348336b',
  };

  return renderType === renderer
    ? renderer.create(<UserName {...requiredProps} />)
    : shallow(<UserName {...requiredProps} />);
};

describe('UserName component', () => {
  it('should render user name as a link', () => {
    const wrapper = renderUserNameComponent(shallow);
    expect(wrapper.find('Link').length).toBe(1);
    expect(wrapper.find('Link').text()).toBe('Henry Harrison');
  });

  it('should render user name as a link', () => {
    const wrapper = renderUserNameComponent(shallow);
    expect(wrapper).toMatchSnapshot();
  });
});
