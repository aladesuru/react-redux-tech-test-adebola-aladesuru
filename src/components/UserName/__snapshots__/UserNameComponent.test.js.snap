// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserName component should render user name as a link 1`] = `
<Link
  to="/userId/e837c9f5-247f-445f-bcc3-7d434348336b"
>
  Henry Harrison
</Link>
`;
